compose/generic-x86.yml

```diff
@@ -6,11 +6,15 @@ services:
   core:
     build: ./core
     network_mode: 'host'
+    privileged: true
     volumes:
       - 'core-storage:/data'
       - 'reports-storage:/reports'
+      - '/var/run/docker.sock:/var/run/docker.sock'
```

Comment:
If we are using the host docker socket we might have trouble accessing the insecure registry on the DUT.

We might want to have dockerode create a daemon instance (inside the core/worker) for us with our specified insecure registries. WDYT @robertgzr ?

Reply:
I'm gonna NACK `privileged: true` unless we have a _very_ good reason.

Reply:
We were thinking it was required for losetup, and now maybe for Docker in Docker, but I agree we should add individual caps where required.

Reply:
Why do we even need access to the host docker socket? I like to comment each privilege/capability as to _why_ it's required. Here's a few examples:
https://github.com/balena-io-playground/balena-gateway-config/blob/36e387bb2c4993c7f78499aaffe80fbbcda3ed2f/docker-compose.yml
https://github.com/jakogut/balena-steam/blob/58d2cb7c75e1705a6a577c53b5f2897ee18d3882/docker-compose.yml

Reply:
We need a docker socket to load and push a docker image for hup tests. I'm thinking we don't want to use the host socket though, but should try for docker-in-docker like we do for preloading.

It will make the image much larger though, so maybe we want to copy the target hostapp to the DUT and load it from there? Avoid needing dockerode/docker on the worker at all?

Reply:
That seems like a cleaner solution to me. I prefer to keep dependencies of tests isolated to that test. Unless it's a generic feature that will be used by many tests, it doesn't make sense to add it to the interface of Leviathan.

Reply:
@klutchell @jakogut we just need to push an image to the registry it's probably simpler to use skopeo like this:

```
skopeo copy --dst-tls-verify=false docker-archive://./image.docker docker://<dut>:5000/hostapp
```

Reply:
@robertgzr pointed out that we might not even need the registry if we can push directly to the docker context on the DUT.